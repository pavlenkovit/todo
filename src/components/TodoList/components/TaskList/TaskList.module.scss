.list {
  flex: 1 1 auto;
  height: 1%;
  overflow: auto;
  margin: 0 -15px;
  padding: 0 15px;
}
