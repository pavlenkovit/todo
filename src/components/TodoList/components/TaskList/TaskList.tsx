import React from 'react';
import styles from './TaskList.module.scss';
import { ITask } from '../../../../models/tasks';
import Task from './components/Task';

export interface ITaskListProps {
  list: ITask[];
}

const TaskList: React.SFC<ITaskListProps> = ({ list }) => {
  return (
    <ul className={styles.list}>
      {list.slice(0).reverse().map((task: ITask) => (
        <Task key={task.id} {...task} />
      ))}
    </ul>
  );
};

export default TaskList;
