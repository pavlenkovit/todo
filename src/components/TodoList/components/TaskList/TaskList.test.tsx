import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TaskList, { ITaskListProps } from './TaskList';

configure({ adapter: new Adapter() });

describe('TaskList', () => {
  const props: ITaskListProps = {
    list: [
      { userId: 1, id: 101, title: 'explicabo enim', completed: false },
      { userId: 1, id: 102, title: 'sed ab consequatur', completed: false },
    ],
  };
  const wrapper = shallow(<TaskList {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('renders 2 task', () => {
    expect(wrapper.find('Connect(Task)')).toHaveLength(2);
  });
});
