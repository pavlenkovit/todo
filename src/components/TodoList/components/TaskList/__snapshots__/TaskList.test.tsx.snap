// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList renders properly 1`] = `ShallowWrapper {}`;
